scripts/aws-cni-support.sh

```diff
@@ -17,57 +17,67 @@
 # Set language to C to make sorting consistent among different environments.
 export LANG=C
 
-set -e
+set -euo pipefail
 LOG_DIR="/var/log/aws-routed-eni"
+mkdir -p ${LOG_DIR}
 
 # collecting L-IPAMD introspection data
-curl http://localhost:61678/v1/enis         > ${LOG_DIR}/eni.output
-curl http://localhost:61678/v1/pods         > ${LOG_DIR}/pod.output
-curl http://localhost:61678/v1/networkutils-env-settings > ${LOG_DIR}/networkutils-env.output
-curl http://localhost:61678/v1/ipamd-env-settings > ${LOG_DIR}/ipamd-env.output
-curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.output
-
-# metrics TODO not able to use LOG_DIR
-curl http://localhost:61678/metrics 2>&1 > /var/log/aws-routed-eni/metrics.output
-
-# collecting kubelet introspection data
-curl http://localhost:10255/pods  > ${LOG_DIR}/kubelet.output
+curl http://localhost:61678/v1/enis         > ${LOG_DIR}/eni.out
+curl http://localhost:61678/v1/pods         > ${LOG_DIR}/pod.out
+curl http://localhost:61678/v1/networkutils-env-settings > ${LOG_DIR}/networkutils-env.out
+curl http://localhost:61678/v1/ipamd-env-settings > ${LOG_DIR}/ipamd-env.out
+curl http://localhost:61678/v1/eni-configs  > ${LOG_DIR}/eni-configs.out
+
+# metrics
+curl http://localhost:61678/metrics 2>&1 > ${LOG_DIR}/metrics.out
+
+# Collecting kubelet introspection data
+if [[ -n "${KUBECONFIG+1}" ]]; then
```

Comment:
Looks like without +1 gives the behavior you want.

Reply:
Fixed!